cohort-engine/src/main/java/com/ibm/cohort/engine/measure/seed/MeasureEvaluationSeeder.java

```diff
@@ -0,0 +1,179 @@
+			LibraryResolutionProvider<Library> libraryResourceProvider) {
+		this.providerFactory = providerFactory;
+		this.libraryLoader = libraryLoader;
+		this.libraryResourceProvider = libraryResourceProvider;
+	}
+
+	public MeasureEvaluationSeeder disableDebugLogging() {
+		this.debugMode = false;
+
+		return this;
+	}
+
+	public MeasureEvaluationSeeder disableExpressionCaching() {
+		this.disableExpressionCaching = true;
+
+		return this;
+	}
+
+
+	public MeasureEvaluationSeeder withTerminologyProvider(String source, String user, String password) {
+		this.terminologyProviderSource = source;
+		this.terminologyProviderUser = user;
+		this.terminologyProviderPassword = password;
+
+		return this;
+	}
+
+
+	public IMeasureEvaluationSeed create(Measure measure, String periodStart, String periodEnd, String productLine) {
+		LibraryHelper.loadLibraries(measure, this.libraryLoader, this.libraryResourceProvider);
+
+		// resolve primary library
+		org.cqframework.cql.elm.execution.Library library =
+                LibraryHelper.resolvePrimaryLibrary(measure, libraryLoader, this.libraryResourceProvider);
+
+		List<Triple<String, String, String>> usingDefs = UsingHelper.getUsingUrlAndVersion(library.getUsings());
+
+		if (usingDefs.size() > 1) {
+			throw new IllegalArgumentException("Evaluation of Measure using multiple Models is not supported at this time.");
+		}
+
+
+		TerminologyProvider terminologyProvider = createTerminologyProvider(usingDefs);
+		LinkedHashMap<Triple<String, String, String>, DataProvider> dataProviders = createDataProviders(usingDefs, terminologyProvider);
+		Interval measurementPeriod = createMeasurePeriod(periodStart, periodEnd);
+		Context context = createContext(library, dataProviders, terminologyProvider, measurementPeriod, productLine);
+
+		List<Map.Entry<Triple<String, String, String>, DataProvider>> dataProviderList = new ArrayList<>(dataProviders.entrySet());
+		DataProvider lastDataProvider = dataProviderList.get(dataProviderList.size() - 1).getValue();
+
+		return new CustomMeasureEvaluationSeed(measure, context, measurementPeriod, lastDataProvider);
+	}
+
+	@VisibleForTesting
+	protected Context createContext(
+			org.cqframework.cql.elm.execution.Library library,
+			Map<Triple<String, String, String>, DataProvider> dataProviders,
+			TerminologyProvider terminologyProvider,
+			Interval measurementPeriod,
+			String productLine) {
+
+		Context context = new Context(library);
+		context.registerLibraryLoader(libraryLoader);
+
+		if (!dataProviders.isEmpty()) {
+			context.registerTerminologyProvider(terminologyProvider);
+		}
+
+		for (Map.Entry<Triple<String, String, String>, DataProvider> dataProviderEntry : dataProviders.entrySet()) {
+			Triple<String, String, String> usingDef = dataProviderEntry.getKey();
+			DataProvider dataProvider = dataProviderEntry.getValue();
+
+			context.registerDataProvider(usingDef.getRight(), dataProvider);
+		}
+
+		context.setParameter(
+		        null,
```

Comment:
Super Pedantic: Is this indented a little too far?
Actually, it looks like this and the following few lines use space indentation instead of what the rest of the file uses.

Reply:
this is also copied code is why that happened fwiw